language/benchmarks/src/move_vm.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use bytecode_verifier::VerifiedModule;
+use criterion::Criterion;
+use libra_state_view::StateView;
+use libra_types::access_path::AccessPath;
+use libra_types::account_address::AccountAddress;
+use libra_types::identifier::{IdentStr, Identifier};
+use libra_types::language_storage::ModuleId;
+use move_lang::shared::Address;
+use move_lang::to_bytecode::translate::CompiledUnit;
+use vm::{
+    gas_schedule::{CostTable, GasAlgebra, GasUnits},
+    transaction_metadata::TransactionMetadata,
+};
+use vm_runtime::{
+    chain_state::TransactionExecutionContext, data_cache::BlockDataCache, move_vm::MoveVM,
+};
+
+/// Entry point for the bench, provide a function name to invoke in Module Bench in bench.move.
+pub fn bench(c: &mut Criterion, fun: &str) {
+    let module = compile_module();
+    let mut move_vm = MoveVM::new();
+    move_vm.cache_module(module);
+    execute(c, &move_vm, fun);
+}
+
+// Compile `bench.move`
+fn compile_module() -> VerifiedModule {
+    // TODO: this has only been tried with `cargo bench` from `libra/src/language/benchmarks`
+    let mut path = std::env::current_dir().unwrap();
+    path.push("src/bench.move");
+    let s: &'static str = Box::leak(Box::new(
+        path.to_str().expect("no path specified").to_owned(),
+    ));
```

Comment:
What's this trying to do? Looks scary.

Reply:
This is due to the current implementation of the source compiler -- particularly its not having a string intern table. @tnowacki can explain better.

Reply:
Yeah, not having an intern table/global memory leads to fun workarounds

I at least cleaned up the API to make all of this internal in #2348

Reply:
Ah I see. Could you add a comment about this?

Reply:
this is gone now after @tnowacki commit